### DISCUSSION TITLE
Query about using web sockets for communication over HTTP

### DISCUSSION THREAD

**Discussion by gkannan1989** [shakacode/react_on_rails#1616] [General]
can standalone @shakacode-tools/react-on-rails-pro-node-renderer communicate with ReactOnRails::ProRendering via web sockets rather than standard HTTP transport?

this one to configure renderer url = config.renderer_url similar like this one for transport? thoughts?

**Comment by justin808** [comment]
@gkannan1989 I need some more information on what you're trying to accomplish.

**Reply by gkannan1989** [reply]
Sure, websockets are the faster way to communicate between internal systems, if we host node renderer as standalone service on our docker container, our ReactOnRails::ProRendering will call this node renderer to process the JS. 

This communication i believe currently over HTTP and react_on_rails_pro library managing this communication, but what we are unsure is there any support for websocket communication rather than HTTP. 

just to improve the response when more requests passed to node renderer

ex: renderer_url = http://xyz.node-renderer.com 

renderer_url being invoked by the library and there is a transport happening to node renderer and back to library over HTTP transport. is there any support for websocket instead of HTTP?

**Comment by justin808** [comment]
We're working on direct communication from the browser to the SSR node server for react server components.

Currently, the renderer supports HTTP and soon to be HTTP2.

Seems like it should be fairly easy: https://www.perplexity.ai/search/can-express-server-4fuTuCq4Qwqlht9uubc3rA

If you use React on Rails Pro, we might help you with that, or you might submit a PR for that.

Does that help?

**Reply by gkannan1989** [reply]
Thanks @justin808 👍 yes it helps.
